R/vaccination.R

```diff
@@ -409,7 +419,34 @@ vaccine_schedule_future <- function(start,
     doses <- mcstate::array_bind(start$doses, doses)
   }
 
-  vaccine_schedule(daily_doses_date, doses)
+  schedule <- vaccine_schedule(daily_doses_date, doses)
+
+  if (!is.null(lag_groups)) {
+    if (is.null(lag_days)) {
+      stop("'lag_days' must be non-NULL if 'lag_groups' is non_NULL")
+    }
+
+    ## we could do this in apply but loop is more readable
+    for (i in lag_groups) {
+      ## original schedule for group i
+      old_schedule_group_i <- schedule$doses[i, , ]
+      nr <- nrow(old_schedule_group_i)
+      nc <- ncol(old_schedule_group_i)
+
+      ## get start of dose schedule for the group
+      start <- which(old_schedule_group_i[1, ] > 0)[1]
```

Comment:
maybe need a catch for this in case this doesn't exist - e.g. people <50 are not eligible so you won't have any value >0 for them